nucypher/blockchain/eth/sol/source/contracts/Issuer.sol

```diff
@@ -160,6 +163,16 @@ contract Issuer is Upgradeable {
         currentSupply2 = currentSupply2.sub(_amount);
     }
 
+    /**
+    * @notice Burn sender's tokens. Amount of tokens will be returned for future minting
+    * @param _value Amount to burn
+    **/
+    function burn(uint256 _value) public {
```

Comment:
```suggestion
    function burn(uint256 _value) public {
    require(currentMintingPeriod <= 100, "Sorry, you waited too long – get some more Nu and start staking!");
```
If the primary purpose of the burn function is to allow Nu holders who don't meet minimum stake size requirements to get their ETH back, then maybe one should only be able to call it up until a predefined moment in time – e.g. within the first 100 periods of the reward schedule

Reply:
Initial idea was for using from worklock and who else wants, but of course we can restrict this method for other Nu holders